### ISSUE TITLE
XHR onload handlers are not being called when onload is overwritten after send.

### ISSUE THREAD

**Issue by jennifer-mann** [cypress-io/cypress#143]
The `onload` handler is never being called in the driver when the app has overwritten `onload` after send. Although the `onreadystatechange` handler *is* called. 

**Example App Code**
```
var xhr = new XMLHttpRequest();

xhr.open('GET', findRoot() + file, true);
xhr.send();

xhr.onload = function () {
  if (xhr.status === 200 && callback) {
    callback(xhr.responseText);
  }
};
```

**Example of Command Log never 404'ing**
![screen shot 2016-05-18 at 1 51 16 pm](https://cloud.githubusercontent.com/assets/1271364/15369300/07dd2cf8-1d00-11e6-8711-592f0f4fdae8.png)

**Comment by brian-mann** [comment]
Fixed in [`0.16.0`](https://github.com/cypress-io/cypress/wiki/changelog#0160-05172016).

**Comment by Mouvedia** [comment]
183b330
